homeassistant/components/here_travel_time/__init__.py

```diff
@@ -1 +1,172 @@
+                RouteMode[TRAFFIC_MODE_ENABLED],
+                arrival,
+                departure,
+            )
+
+            response: RoutingResponse = self._api.public_transport_timetable(
+                here_formatted_origin,
+                here_formatted_destination,
+                True,
+                [
+                    RouteMode[self.config.route_mode],
+                    RouteMode[self.config.travel_mode],
+                    RouteMode[TRAFFIC_MODE_ENABLED],
+                ],
+                arrival=arrival,
+                departure=departure,
+            )
+
+            _LOGGER.debug(
+                "Raw response is: %s", response.response  # pylint: disable=no-member
+            )
+
+            source_attribution = response.response.get(  # pylint: disable=no-member
+                "sourceAttribution"
+            )
+            attribution: str | None = None
+            if source_attribution is not None:
+                attribution = build_hass_attribution(source_attribution)
+            route: list = response.response["route"]  # pylint: disable=no-member
+            summary: dict = route[0]["summary"]
+            waypoint: list = route[0]["waypoint"]
+            distance: float = summary["distance"]
+            traffic_time: float = summary["baseTime"]
+            if self.config.travel_mode in TRAVEL_MODES_VEHICLE:
+                traffic_time = summary["trafficTime"]
+            if self.config.units == CONF_UNIT_SYSTEM_IMPERIAL:
+                # Convert to miles.
+                distance = distance / 1609.344
+            else:
+                # Convert to kilometers
+                distance = distance / 1000
+            return {
+                ATTR_ATTRIBUTION: attribution,
+                ATTR_DURATION: summary["baseTime"] / 60,
+                ATTR_DURATION_IN_TRAFFIC: traffic_time / 60,
+                ATTR_DISTANCE: distance,
+                ATTR_ROUTE: response.route_short,
+                ATTR_ORIGIN: ",".join(here_formatted_origin),
+                ATTR_DESTINATION: ",".join(here_formatted_destination),
+                ATTR_ORIGIN_NAME: waypoint[0]["mappedRoadName"],
+                ATTR_DESTINATION_NAME: waypoint[1]["mappedRoadName"],
+            }
+        return None
+
+
+def build_hass_attribution(source_attribution: dict) -> str | None:
+    """Build a hass frontend ready string out of the sourceAttribution."""
+    if (suppliers := source_attribution.get("supplier")) is not None:
+        supplier_titles = []
+        for supplier in suppliers:
+            if (title := supplier.get("title")) is not None:
+                supplier_titles.append(title)
+        joined_supplier_titles = ",".join(supplier_titles)
+        attribution = f"With the support of {joined_supplier_titles}. All information is provided without warranty of any kind."
+        return attribution
+    return None
+
+
+def convert_time_to_isodate(timestr: str) -> str | None:
```

Comment:
I see the logic in this function changed while it was moving between files. I would suggest this whole thing would be much easier to review if it wasn't changing logic and moving between files at the same time.  Is that possible?

Similarly, the test refactorings make it hard to see what changed vs what is just a cleanup.

So maybe separate cleanups into pre or post factorings?

Reply:
I did refactor this method and can revert it as described in the other comments.
However I am missing where the logic changed. Can you point me to it @allenporter ?

Reply:
Nevermind I spotted the issue!